### ISSUE TITLE
Mismatch of allowed spaces in namespaced map syntax

### ISSUE THREAD

**Issue by NoahTheDuke** [borkdude/edamame#98]
**version**

1.3.20 + 1.3.21

**platform**

Unix

**problem**

In 1.3.20, `#:: {:a 1}` was disallowed as auto-resolved namespaced maps were expected to have no whitespace.

In 1.3.21, the rules were loosened beyond what Clojure allows, accepting forms that throw errors in Clojure.

**repro**

1.3.20:

```
user=> (e/parse-string "#:: {:a 1}" '{:auto-resolve {:current user f foo}})
Execution error (ExceptionInfo) at edamame.impl.parser/throw-reader (parser.cljc:44).
Invalid symbol:
user=> #:: {:a 1}
{:user/a 1}
```

I won't show the rest as they're all the same error.

1.3.21:

```
user=> (e/parse-string "#:: {:a 1}" '{:auto-resolve {:current user f foo}})
{:user/a 1}
user=> #:: {:a 1}
{:user/a 1}

user=> (e/parse-string "#: :{:a 1}" '{:auto-resolve {:current user f foo}})
{:user/a 1}
user=> #: :{:a 1}
Syntax error reading source at (REPL:1:3).
Namespaced map must specify a namespace
Syntax error reading source at (REPL:1:5).
Invalid token: :
{:a 1}

user=> (e/parse-string "#:: f {:a 1}" '{:auto-resolve {:current user f foo}})
{:foo/a 1}
user=> #:: f {:a 1}
Syntax error reading source at (REPL:1:5).
Namespaced map must specify a namespace
Syntax error compiling at (REPL:0:0).
Unable to resolve symbol: f in this context
{:a 1}

user=> (e/parse-string "#: :f {:a 1}" '{:auto-resolve {:current user f foo}})
{:foo/a 1}
user=> #: :f {:a 1}
Syntax error reading source at (REPL:1:3).
Namespaced map must specify a namespace
:f
{:a 1}

user=> (e/parse-string "#: : {:a 1}" '{:auto-resolve {:current user f foo}})
{:user/a 1}
user=> #: : {:a 1}
Syntax error reading source at (REPL:1:3).
Namespaced map must specify a namespace
Syntax error reading source at (REPL:1:5).
Invalid token: :
{:a 1}

user=> (e/parse-string "#: : f {:a 1}" '{:auto-resolve {:current user f foo}})
{:foo/a 1}
user=> #: : f {:a 1}
Syntax error reading source at (REPL:1:3).
Namespaced map must specify a namespace
Syntax error reading source at (REPL:1:5).
Invalid token: :
Syntax error compiling at (REPL:0:0).
Unable to resolve symbol: f in this context
{:a 1}

```

**expected behavior**

I expect that the parsing behavior matches Clojure: `#:: {:a 1}` and `#::f {:a 1}` but not any other kinds of whitespace (between the first and second `:` or between the second `:` and the alias symbol).

**Comment by NoahTheDuke** [comment]
The change in #97 handled this to match Clojure's reader, but there may be other issues I don't know about.

**Comment by borkdude** [comment]
Fixed on master. The fix in #97 wasn't bad but I added some stricter checking so exception are caught with better error messages.

**Comment by NoahTheDuke** [comment]
Thank you, appreciate the quick response.
